source/geod24/concurrency.d

```diff
@@ -815,6 +820,39 @@ public class InfoThread : Thread
 }
 
 
+private FiberScheduler thisSchedulerStorage;
+
+/***************************************************************************
+
+    Get current running FiberScheduler
+
+    Returns:
+        Returns a reference to the current scheduler or null no
+        scheduler is running
+
+    TODO: Support for nested schedulers
+
+***************************************************************************/
+
+public FiberScheduler thisScheduler () nothrow
+{
+    return thisSchedulerStorage;
+}
+
+/***************************************************************************
+
+    Set current running FiberScheduler
+
+    Params:
+        value = Reference to the current FiberScheduler
+
+***************************************************************************/
+
+public void thisScheduler (FiberScheduler value) nothrow
```

Comment:
I guess you could mark this as a `@property` and then use it like `thisScheduler = null` for example. Because `thisScheduler(arg)` looks really strange.

Reply:
They were initially marked as a property but I ran into a problem. Will check it again.

Reply:
I wonder if it's something to do with overloads.

Well, `@property` is a bit weird because it's got some edge-cases. You might have ran into them.